proxygen_cli/cli/command_main.py

```diff
@@ -9,15 +10,30 @@
     command_spec,
     command_secret,
     command_docker,
-    command_pytest_nhsd_apim_token
+    command_pytest_nhsd_apim_token,
 )
 
 
+# Fetch the version from pyproject.toml
+def get_version():
+    pyproject_path = Path(__file__).resolve().parent.parent.parent / "pyproject.toml"
+    if pyproject_path.exists():
+        pyproject = toml.load(pyproject_path)
+        return pyproject.get("tool", {}).get("poetry", {}).get("version", "unknown")
+    return "unknown"
+
```

Comment:
Instead of a custom function to parse the pyproject.toml, looks like click has a version_option decorator you can add to the main function  https://click.palletsprojects.com/en/stable/api/#click.version_option, meaning the change would be
```
@click.group()
@click.version_option()
def main():
    """Main group"""
```
As we already have the `importlib_metadata` package installed, this should work
Then can call `proxygen --version` instead which seems more standard

This may change the test a bit

Reply:
Initially I have implemented the same, this is causing a conflict while writing the test case, which has made to me write a separate function

Reply:
Updated accordingly